webapp/src/components/tool_approval_set.tsx

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2023-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState} from 'react';
+import styled from 'styled-components';
+
+import {doToolCall} from '@/client';
+
+import {ToolCall, ToolCallStatus} from './llmbot_post';
+import ToolCard from './tool_card';
+
+// Styled components
+const ToolCallsContainer = styled.div`
+    display: flex;
+    flex-direction: column;
+    gap: 8px;
+    margin: 16px 0;
+`;
+
+// Tool call interfaces
+interface ToolApprovalSetProps {
+    postID: string;
+    toolCalls: ToolCall[];
+}
+
+const ToolApprovalSet: React.FC<ToolApprovalSetProps> = (props) => {
+    // Track which tools are currently being processed
+    const [processingTools, setProcessingTools] = useState<string[]>([]);
+    const [error, setError] = useState('');
+    const [collapsedTools, setCollapsedTools] = useState<string[]>([]);
+
+    const handleApprove = async (toolID: string) => {
+        try {
+            // Update UI immediately to prevent double-clicking
+            setProcessingTools((prev) => [...prev, toolID]);
+
+            // Call the API endpoint
+            await doToolCall(props.postID, [toolID]);
+        } catch (err) {
+            setError('Failed to approve tool call');
+
+            // Remove from processing state on error
+            setProcessingTools((prev) => prev.filter((id) => id !== toolID));
+        }
+    };
+
+    const handleReject = async (toolID: string) => {
+        try {
+            // Update UI immediately to prevent double-clicking
+            setProcessingTools((prev) => [...prev, toolID]);
+
+            // Call the API endpoint
+            await doToolCall(props.postID, []);
```

Comment:
I don't know if that currently can happen but if you have multiple tools requesting for permission at the same time, all gets rejected. Also, if you have multiple of them, and you accept 1 of them, you are rejecting the others. I'm not sure if that is the case, but that is my understanding from the  code.

Reply:
Fixed.